hal/src/boron/network/sara_ncp_client.cpp

```diff
@@ -498,52 +498,59 @@ int SaraNcpClient::getSignalQuality(CellularSignalQuality* qual) {
     CHECK(checkParser());
     CHECK(queryAndParseAtCops(qual));
 
+    // Min and max RSRQ index values multiplied by 100
+    // Min: -19.5 and max: -3
+    const int min_rsrq_mul_by_100 = -1950;
+    const int max_rsrq_mul_by_100 = -300;
+
     if (ncpId() == PLATFORM_NCP_SARA_R410) {
-        int rxlev, rxqual, rscp, ecn0, rsrq, rsrp;
-        auto resp = parser_.sendCommand("AT+CESQ");
-        int r = CHECK_PARSER(resp.scanf("+CESQ: %d,%d,%d,%d,%d,%d", &rxlev, &rxqual,
-                &rscp, &ecn0, &rsrq, &rsrp));
-        CHECK_TRUE(r == 6, SYSTEM_ERROR_AT_RESPONSE_UNEXPECTED);
-        r = CHECK_PARSER(resp.readResult());
-        CHECK_TRUE(r == AtResponse::OK, SYSTEM_ERROR_AT_NOT_OK);
+        int rsrp;
+        int rsrq_n;
+        unsigned long rsrq_f;
+
+        // Default to 255 in case RSRP/Q are not found
+        qual->strength(255);
+        qual->quality(255);
+
+        // Set UCGED to mode 5 for RSRP/RSRQ values on R410M
+        CHECK_PARSER_OK(parser_.execCommand("AT+UCGED=5"));
+        auto resp = parser_.sendCommand("AT+UCGED?");
+        int r1 = 0;
+        int r2 = 0;
+        while (resp.hasNextLine()) {
+            r1 = CHECK_PARSER(resp.scanf("+RSRP: %*d,%*d,\"%d.%*u\"", &rsrp));
```

Comment:
This will not work correctly, because underneath `resp.scanf()` will call `readLine()`. So, these two calls will read two lines, whereas the intended behavior is supposed to be to try to parse either RSRP or RSRQ for the line. `readLine()` + `sscanf()` should be used or a single `resp.scanf()` call that can parse both RSRP/RSRQ (I gave an example earlier).